app/controllers/providers_controller.rb

```diff
@@ -1,11 +1,13 @@
 class ProvidersController < ApplicationController
+  rescue_from JsonApiClient::Errors::NotFound, with: :not_found
+
   def index
     @providers = Provider.all
     render_manage_ui if @providers.empty?
     redirect_to provider_path(@providers.first.provider_code) if @providers.size == 1
   end
 
   def show
-    @provider = Provider.find(params[:code]).first.attributes
+    @provider = Provider.find(params[:code]).first
```

Comment:
The exception wasn’t being caught in the action so had todo it at controller level.